core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/SigningService.java

```diff
@@ -89,6 +95,30 @@ public String upcheck() {
 
   @JsonRpcMethod("Filecoin.WalletList")
   public List<String> filecoinWalletList() {
-    return fcKeyIdentifiers.list(KeyType.SECP256K1);
+    return fcSigners.availableIdentifiers().parallelStream().collect(Collectors.toList());
+  }
+
+  @JsonRpcMethod("Filecoin.WalletSign")
+  public Bytes filecoinWalletSign(final String filecoinAddress, final String dataToSign) {
+    // convert dataToSign from Base32 to Bytes
```

Comment:
This comment isn't quite right, it's only the address that is in base32, the data is in base64

Reply:
removed